crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/CacheChecker.java

```diff
@@ -28,22 +30,21 @@
   private final Cache cache;
 
   /**
-   * @return the last modified time for the file. Recursively finds the largest last modified time
-   *     for all subfiles if {@code file} is a directory.
+   * @return the last modified time for the file. Recursively finds the most recent last modified
+   *     time for all subfiles if {@code file} is a directory.
    */
   private static long getLastModifiedTime(File file) throws IOException {
     long lastModifiedTime = file.lastModified();
 
-    if (file.isDirectory()) {
-      File[] subFiles = file.listFiles();
-      if (subFiles == null) {
-        throw new IOException("Failed to read directory: " + file.getAbsolutePath());
+    if (file.canRead()) {
+      OptionalLong maxLastModifiedTime =
+          Files.walk(file.toPath()).mapToLong(path -> path.toFile().lastModified()).max();
```

Comment:
You don't need to convert to a file here. Just use Files.getLastModifiedTime(path)

In general converting to a java.io.File should not be necessary except when interfacing with older third party APIs (e.g. Eclipse APIs) that assume the use of java.io.File. All standard operations in the JDK should be able to be done with a. Path.